cmd/metrics-v3-cache.go

```diff
@@ -108,14 +136,40 @@ func newDriveMetricsCache() *cachevalue.Cache[storageMetrics] {
 		storageInfo := objLayer.LocalStorageInfo(GlobalContext, true)
 		onlineDrives, offlineDrives := getOnlineOfflineDisksStats(storageInfo.Disks)
 		totalDrives := onlineDrives.Merge(offlineDrives)
-		v = storageMetrics{
-			storageInfo:   storageInfo,
-			onlineDrives:  onlineDrives.Sum(),
-			offlineDrives: offlineDrives.Sum(),
-			totalDrives:   totalDrives.Sum(),
+
+		currentStats := getCurrentDriveIOStats()
+		now := time.Now().UTC()
+
+		prevDriveIOStatsMu.Lock()
+		if prevDriveIOStats != nil {
+			duration := now.Sub(prevDriveIOStatsRefreshedAt)
+			if duration.Seconds() > 1 {
+				diffStats := map[string]madmin.DiskIOStats{}
+				for d, cs := range currentStats {
+					if ps, found := prevDriveIOStats[d]; found {
+						diffStats[d] = getDiffStats(ps, cs)
+					}
+				}
+
+				ioStats := &diskIOStats{stats: diffStats, duration: duration}
+
+				v = storageMetrics{
+					storageInfo:   storageInfo,
+					onlineDrives:  onlineDrives.Sum(),
+					offlineDrives: offlineDrives.Sum(),
+					totalDrives:   totalDrives.Sum(),
```

Comment:
These values will be set only if `prevDriveIOStats != nil`. Can we move this out of the `if` block?

Reply:
thanks, done.